src/curator/frontend_utils.py

```
@@ -0,0 +1,73 @@
+            tmp_file.flush() # Ensure all data is written to disk
+            tmp_file.seek(0) # Rewind the file to the beginning for reading by tarfile
+
+            if not tarfile.is_tarfile(tmp_file):
+                tmp_file.seek(0) # Rewind again for debug read
+                print(f"Error: Downloaded data is not a valid tarfile.")
+                print(f"First 100 bytes of the downloaded data: {tmp_file.read(100)}")
+                sys.exit(1)
+
+            tmp_file.seek(0) # Rewind again before extraction
+            with tarfile.open(fileobj=tmp_file, mode="r:gz") as tar: # Use the file object directly
+                tar.extractall(path=target_dir)
```

Ruff (0.11.9)

Undefined name `target_dir`

(F821)